utils/FileCache.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Handles this tool's disk cache, maintaining a separate directory and cleaning files from it
+ */
+class FileCache {
+	// Cleanup temporary storage at the end of 5% of requests
+	private const CLEANUP_RATE = 5;
+
+	// Delete files after this amount of time in seconds
+	private const CACHE_DURATION = 24 * 60 * 60;
+
+	/** @var string Tool-specific temporary directory */
+	private $dir;
+
+	/** @var FileCache */
+	private static $instance;
+
+	/**
+	 * @param string $path Directory for temp files
+	 */
+	private function __construct( string $path ) {
+		$this->dir = $this->makePrivateDirectory( $path );
+
+		// Randomly clean up the cache
+		if ( mt_rand( 0, 100 ) <= self::CLEANUP_RATE ) {
+			register_shutdown_function( function() {
+				$this->cleanup();
+			} );
+		}
+	}
+
+	/**
+	 * Returns a global instance of this class
+	 * @param array|null $config Tool configuration
+	 * @return FileCache
+	 */
+	public static function singleton( array $config = null ): FileCache {
+		if ( !self::$instance ) {
+			if ( !$config ) {
+				throw new LogicException( 'No configuration provided for the first call to ' . __METHOD__ );
+			}
+			self::$instance = new FileCache( $config['tempPath'] );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * @return string Directory for temporary files
+	 */
+	public function getDirectory(): string {
+		return $this->dir;
+	}
+
+	/**
+	 * Creates a tool-specific temporary directory with specific permissions
+	 *
+	 * @param string $path
+	 * @return string
+	 */
+	private function makePrivateDirectory( string $path ): string {
+		$dir = $path;
+
+		// Use username in the path. The tool needs its own directory so that it knows which
+		// files to garbage collect. Can't just use the hardcoded name because
+		// wsexport and wsexport-test might end up on the same node, stepping on each other's toes.
+		$user = get_current_user();
+		// Remove leading "tools." present if running on labs
+		$userCleaned = preg_replace( '/^tools\./', '', $user );
+		if ( $user !== $userCleaned ) {
+			// We're on Toolforge, gotta compartmentalize
+			$dir = $path . '/' . $userCleaned;
+		}
```

Comment:
Is it important of doing it only on labs? We might want to do it everywhere to make code simpler.

Reply:
Done.